fix: 抽選システム設定保存で duplicate key 違反（古い単独 unique 制約の残存）

## バグの概要

抽選のシステム設定画面で「締め切りなし」にチェックを入れて保存すると、`保存に失敗しました` と表示され保存できない。

サーバーログ:
```
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "uknm18l4pyovtvd8y3b3x0l2y64"
```

## 再現手順

1. SUPER_ADMIN または ADMIN でログイン
2. システム設定画面で**設定がまだ保存されていない団体**を選び、抽選締め切り欄を変更（締め切りなしチェック等）
3. 保存

→ 別団体に同じ `setting_key`（例: `lottery_deadline_days_before`）の行が既に存在する場合、INSERT が単独 unique 制約に違反する。

## 期待される動作

団体ごとに同じ `setting_key` を保存できる（複合 unique は `(setting_key, organization_id)`）。

## 実際の動作

`uknm18l4pyovtvd8y3b3x0l2y64` という古い `(setting_key)` 単独 unique 制約が本番DBに残っており、別団体での新規 INSERT が失敗する。

## 原因

`add_cancel_reason_and_system_settings.sql` が初期に `setting_key VARCHAR(100) NOT NULL UNIQUE` で作成した制約は、Hibernate 自動生成名 `uknm18l4pyovtvd8y3b3x0l2y64` で本番に存在。
`migration_organization.sql` は `system_settings_setting_key_key` / `uk_system_settings_setting_key` の名前で DROP を試みたが実際の名前と一致せず、結果として単独 unique 制約が残ったまま `(setting_key, organization_id)` 複合 unique が追加された。

本番DBの `\d system_settings`:
```
"uk3uukkredspdgpq65rggt9l5m4" UNIQUE CONSTRAINT, btree (setting_key, organization_id)
"uk_system_settings_key_org"  UNIQUE CONSTRAINT, btree (setting_key, organization_id)
"uknm18l4pyovtvd8y3b3x0l2y64" UNIQUE CONSTRAINT, btree (setting_key)   ← これが原因
```

## 修正方針

新規マイグレーション SQL で:
- `uknm18l4pyovtvd8y3b3x0l2y64` を DROP（本来の修正）
- 重複している `uk3uukkredspdgpq65rggt9l5m4` も DROP（`uk_system_settings_key_org` と完全に同じ列構成のため）

CLAUDE.md の DBマイグレーション適用ルールに従い、本番DBにも psql で適用する。

--- a/database/drop_legacy_system_settings_unique.sql
+++ b/database/drop_legacy_system_settings_unique.sql
@@ -1,0 +1,21 @@
+-- ============================================
+-- system_settings の不要な unique 制約を削除
+-- 関連Issue: #557
+-- ============================================
+--
+-- 経緯:
+--   - 初期版で `setting_key VARCHAR(100) NOT NULL UNIQUE` で作成した単独 unique 制約が
+--     Hibernate 自動生成名 `uknm18l4pyovtvd8y3b3x0l2y64` で本番DBに残存していた。
+--   - `migration_organization.sql` で `system_settings_setting_key_key` /
+--     `uk_system_settings_setting_key` の名前で DROP を試みたが実際の名前と一致せず、
+--     `(setting_key, organization_id)` 複合 unique を追加した結果、単独 unique も生き残った。
+--   - これにより別団体で同じ setting_key を保存しようとすると重複違反になる。
+--
+-- また、複合 unique も `uk_system_settings_key_org` と Hibernate 自動生成名
+-- `uk3uukkredspdgpq65rggt9l5m4` の2つに重複しているため自動生成名側を削除する。
+
+ALTER TABLE system_settings
+  DROP CONSTRAINT IF EXISTS uknm18l4pyovtvd8y3b3x0l2y64;
+
+ALTER TABLE system_settings
+  DROP CONSTRAINT IF EXISTS uk3uukkredspdgpq65rggt9l5m4;

--- a/docs/DESIGN.md
+++ b/docs/DESIGN.md
@@ -314,10 +314,13 @@ Entity Layer (JPA Entity)
 | カラム名 | 型 | 制約 | 説明 |
 |---------|-----|------|------|
 | id | BIGINT | PK, AUTO_INCREMENT | 設定ID |
-| setting_key | VARCHAR(100) | NOT NULL, UNIQUE | 設定キー |
+| setting_key | VARCHAR(100) | NOT NULL | 設定キー |
 | setting_value | VARCHAR(255) | NOT NULL | 設定値 |
+| organization_id | BIGINT | NOT NULL, FK | 団体ID |
 | updated_at | DATETIME | NOT NULL | 更新日時 |
 | updated_by | BIGINT | | 更新者ID |
+
+一意制約: `(setting_key, organization_id)`
 
 **初期データ**:
 - `lottery_deadline_days_before` = `0`（締切日数：月初から何日前。0=前月末日の0時）

--- a/docs/SPECIFICATION.md
+++ b/docs/SPECIFICATION.md
@@ -1527,10 +1527,13 @@ venues ──< venue_match_schedules (venueId)
 | カラム | 型 | 制約 | 説明 |
 |---|---|---|---|
 | id | BIGINT | PK, AUTO | — |
-| setting_key | VARCHAR(100) | NOT NULL, UNIQUE | 設定キー |
+| setting_key | VARCHAR(100) | NOT NULL | 設定キー |
 | setting_value | VARCHAR(255) | NOT NULL | 設定値 |
+| organization_id | BIGINT | NOT NULL, FK | 団体ID |
 | updated_at | TIMESTAMP | NOT NULL | — |
 | updated_by | BIGINT | — | 更新者ID |
+
+一意制約: `(setting_key, organization_id)`
 
 初期データ: `lottery_deadline_days_before` = `0`
 

